site/docs/api/serialization.md

```diff
@@ -321,3 +321,56 @@ a jar file. Moreover, used type ids have to be unique across all
 serializers. Despite those limitations cluster level serializers offer
 full support for IMDG [sources and sinks](sources-sinks.md) - in particular
 the possibility to query & update `Map`s as well as read from `EventJournal`.
+
+### 3rd Party Serialization Support
+
+#### Google Protocol Buffers
+
+As Google Protocol Buffers types already implement `Serializable` interface
+they are natively supported and don't need custom serializer. However, for
+best performance we encourage using Protocol Buffers specific serialization.
+Jet extensions already contain an adapter and hook (supporting Protocol
+Buffers v3) that allow you to make use of it.
+
+To include Maven dependency add following to your pom:
+
+```xml
+<dependency>
+    <groupId>com.hazelcast.jet</groupId>
+    <artifactId>hazelcast-jet-protobuf</artifactId>
+    <classifier>jar-with-dependencies</classifier>
+    <version>${hazelcast.jet.version}</version>
+</dependency>
+```
+
+If you want to use it with just a job, implement provided adapter:
```

Comment:
do you need to add any additional classes ? (Protobuf specific) or just registering serializer is enough, without calling addClass ?

Reply:
You need to `addJar()` the extension obviously.

Reply:
this should be in the docs

Reply:
Yes, it shades the protobuf lib.